ledger/light/executor/filaments.go

```diff
@@ -180,12 +192,39 @@ func (m *FilamentModifierDefault) SetRequest(
 		return nil, nil, errors.New("request from the past")
 	}
 
+	// if request has duplicate or result already exists, return RequestDuplicate result
 	foundRequest, foundResult, err := m.calculator.RequestDuplicate(ctx, requestID.Pulse(), objectID, requestID, request)
 	if err != nil {
 		return nil, nil, err
 	}
 	if foundRequest != nil || foundResult != nil {
-		return foundRequest, foundResult, err
+		return foundRequest, foundResult, nil
+	}
+
+	// check outgoing request if any pending exists with the same ID as request's reason.
+	isAnyPendingWithOutgoingRequestReason := func(req record.Request) error {
+		pendings, err := m.calculator.PendingRequests(ctx, requestID.Pulse(), objectID)
+		if err != nil {
+			return err
+		}
+		reason := req.ReasonRef()
+		for _, p := range pendings {
+			if p == *reason.Record() {
+				return nil
+			}
+		}
+		return errors.Errorf("object ID %v not found in pendings for request %v", objectID, requestID)
```

Comment:
`return errors.Errorf("reason ID %v not found in pendings for object %v", reason.Record().DebugString(), objectID.DebugString())`